server/server/Controllers/CatalogController.cs

```diff
@@ -0,0 +1,117 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.AspNetCore.Mvc;
+using server.Models;
+using System.IO;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.EntityFrameworkCore;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using server.Models.HelpModels;
+using server.Interfaces;
+using server.DBRepositories;
+
+namespace server.Controllers
+{
+    [Route("api/[controller]")]
+    public class CatalogController : Controller
+    {
+        private UnitOfWork db;
+        public CatalogController(ApplicationContext context)
+        {
+            db = new UnitOfWork(context);
+        }
+
+        [HttpGet]
+        public CatalogWithTotalCount Get(int page, int pageSize)
+        {
+            int catalogLength = db.Catalog.GetAll().Count();
+
+            if (pageSize == 0)
+            {
+                pageSize = catalogLength;
+            }
+
+            List<Cake> catalog = db.Catalog.GetAll().ToList();
+            var catalogArray = catalog.ToArray();
+
+            int length = catalogArray.Length;
+            int startIndex = (page - 1) * pageSize;
```

Comment:
Better to implement paging with Linq methods .Skip(pageNumber*pageSize).Take(pageSize)